### PR TITLE
tests: fix system_spec when run with clipboard manager

### LINKED ISSUE #4900: Tests which access the X selections hang if something is asserting ownership
- `nvim --version`: HEAD
- Vim (version: NA) behaves differently?
- Operating system/version: Debian sid
### Actual behaviour

If a program is currently asserting ownership of an X selection used by a test, the test will hang due to xsel/xclip not exiting.
### Expected behaviour

The test completes successfully, regardless of whether the selection is asserted when the test runs.
### Steps to reproduce using `nvim -u NORC`
- Double-click on a word in a terminal, thus asserting the PRIMARY selection.
- make functionaltest TEST_FILE=test/functional/legacy/eval_spec.lua

I ran into this while investigating the QB test failures for #4841, although I'm not entirely sure that it's related since I wouldn't expect any selections to be asserted there.

### PR DESCRIPTION
**Pull request by blueyed**
Replaces "xclip" with a dedicated helper program.

 Fixes: https://github.com/neovim/neovim/issues/4900#issuecomment-501866842

### CODE REVIEW

File: test/functional/eval/system_spec.lua [hunk 1]
```diff
@@ -379,14 +372,11 @@ describe('system()', function()
...
-      it('will exit properly after passing input', function()
-        eq('', eval([[system('xclip -i -loops 1 -selection clipboard', 'clip-data')]]))
-        eq('clip-data', eval([[system('xclip -o -selection clipboard')]]))
-      end)
-    end
+    it('will exit properly after passing input', function()
+      local out = eval(string.format("system('%s', 'clip-data')", nvim_dir..'/streams-test'))
+      assert(out:sub(0, 5) == 'pid: ', out)
+      os_kill(out:match("%d+"))
+    end)
```
Comment (justinmk) [inline test/functional/eval/system_spec.lua]: the describe() scope is not needed, can just be it().

**Comment by justinmk** [comment]
nice change! Doesn't build on MSVC though.

**Comment by blueyed** [comment]
> Doesn't build on Windows though.

The main problem is the absence of `fork()`.
Should we use [CreateThread](https://docs.microsoft.com/en-us/windows/win32/api/processthreadsapi/nf-processthreadsapi-createthread) for it maybe?
Not sure if that would even exit with stdout being kept open then in the first place though.

Before this it was likely skipped on Windows (due to missing "xclip") I assume.

/cc @janlazo @erw7

**Comment by justinmk** [comment]
> Before this it was likely skipped on Windows (due to missing "xclip") I assume.

yes, Windows doesn't have xclip.

> The main problem is the absence of `fork()`.
> Should we use [CreateThread](https://docs.microsoft.com/en-us/windows/win32/api/processthreadsapi/nf-processthreadsapi-createthread) for it maybe?

Either skip it or maybe use libuv.

**Comment by blueyed** [comment]
> maybe use libuv.

I've tried to use threads via libuv, but that would not exit on Linux then also.

I've found a similar test in libuv itself (using `fork()`, https://github.com/libuv/libuv/blob/v1.x/test/test-pipe-close-stdout-read-stdin.c), but it is skipped there for Windows also.

**Review by justinmk — APPROVED** [review]

**Pull request merged**